lib/platform/utils/pr-body.ts

```diff
@@ -0,0 +1,22 @@
+import { appName } from '../../config/app-strings';
+
+export function smartTruncate(input: string, len: number): string {
+  if (input.length < len) {
+    return input;
+  }
+  const releaseNotesMatch = input.match(
+    new RegExp(`### Release Notes.*### ${appName} configuration`, 'ms')
+  );
+  if (releaseNotesMatch) {
+    const divider = `</details>\n\n---\n\n### ${appName} configuration`;
+    const [releaseNotes] = releaseNotesMatch;
+    const nonReleaseNotesLength =
+      input.length - releaseNotes.length - divider.length;
+    const availableLength = len - nonReleaseNotesLength;
```

Comment:
@rarkins Should we catch possible negative `availableLength` ?

Reply:
Yes, good idea

Reply:
What should we do then? No truncate or fallback to simple substring?

Reply:
I think fall back to substring. It's a pretty hopeless case anyway!

Reply:
done